Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -1040,5 +1304,260 @@ public void TestGetQueryId()
+                    Assert.AreEqual(QueryStatus.Success, cmd.GetQueryStatus(queryId));
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test, NonParallelizable]
+        public void TestExecuteNormalQueryWhileAsyncExecQueryIsRunning()
+        {
+            string queryId;
+            var expectedWaitTime = 5;
+
+            SnowflakeDbConnection[] connections = new SnowflakeDbConnection[3];
+            for (int i = 0; i < connections.Length; i++)
+            {
+                connections[i] = new SnowflakeDbConnection(ConnectionString);
+                connections[i].Open();
+            }
+
+            // Start the async exec query
+            using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)connections[0].CreateCommand())
+            {
+                // Arrange
+                cmd.CommandText = $"CALL SYSTEM$WAIT({expectedWaitTime}, \'SECONDS\');";
+
+                // Act
+                queryId = cmd.ExecuteInAsyncMode();
+
+                // Assert
+                Assert.IsTrue(QueryStatuses.IsStillRunning(cmd.GetQueryStatus(queryId)));
+            }
+
+            // Execute a normal query
+            using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)connections[1].CreateCommand())
+            {
+                // Arrange
+                cmd.CommandText = $"select 1;";
+
+                // Act
+                var row = cmd.ExecuteScalar();
+
+                // Assert
+                Assert.AreEqual(1, row);
+            }
+
+            // Get results of the async exec query
+            using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)connections[2].CreateCommand())
+            {
+                // Act
+                DbDataReader reader = cmd.GetResultsFromQueryId(queryId);
+
+                // Assert
+                Assert.IsTrue(reader.Read());
+                Assert.AreEqual($"waited {expectedWaitTime} seconds", reader.GetString(0));
+                Assert.AreEqual(QueryStatus.Success, cmd.GetQueryStatus(queryId));
+            }
+
+            for (int i = 0; i < connections.Length; i++)
+            {
+                connections[i].Close();
+            }
+        }
+
+        [Test]
+        public void TestFailedAsyncExecQueryThrowsError()
+        {
+            string queryId;
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Arrange
+                    var statusRetryCount = 0;
+                    cmd.CommandText = $"SELECT * FROM FAKE_TABLE;";
+
+                    // Act
+                    queryId = cmd.ExecuteInAsyncMode();
+                    while (statusRetryCount < 5 && QueryStatuses.IsStillRunning(cmd.GetQueryStatus(queryId)))
```

Comment:
let's create `var statusMaxRetryCount = 5` in the arrange section and use the variable here

Reply:
Done